snapcraft/_store.py

```diff
@@ -272,12 +274,99 @@ def download(snap_name, channel, download_path, arch):
     try:
         with _requires_login():
             store.download(snap_name, channel, download_path, arch)
-    except storeapi.errors.SnapNotFoundError:
-        raise RuntimeError(
-            'Snap {name} for {arch} cannot be found'
-            ' in the {channel} channel'.format(name=snap_name, arch=arch,
-                                               channel=channel))
     except storeapi.errors.SHAMismatchError:
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def gated(snap_name):
+    """Print list of snaps gated by snap_name."""
+    store = storeapi.StoreClient()
+    # Get data for the gating snap
+    with _requires_login():
+        snap_data = store.cpi.get_package(snap_name, 'stable', None)
```

Comment:
I think we should resolve the snap-id via account-info (logged users accessible snaps) instead of hitting CPI and require an active publication in 'stable'.

Also 'None' is not needed, arch is optional